extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/multiplepersistenceunits/MultiplePersistenceUnitsCdiSessionTest.java

```diff
@@ -0,0 +1,79 @@
+package io.quarkus.hibernate.orm.multiplepersistenceunits;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+
+import org.hibernate.Session;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.hibernate.orm.PersistenceUnit;
+import io.quarkus.hibernate.orm.multiplepersistenceunits.model.config.DefaultEntity;
+import io.quarkus.hibernate.orm.multiplepersistenceunits.model.config.inventory.Plane;
+import io.quarkus.hibernate.orm.multiplepersistenceunits.model.config.user.User;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class MultiplePersistenceUnitsCdiSessionTest {
+
+    @RegisterExtension
+    static QuarkusUnitTest runner = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClass(DefaultEntity.class)
+                    .addClass(User.class)
+                    .addClass(Plane.class)
+                    .addAsResource("application-multiple-persistence-units.properties", "application.properties"));
+
+    @Inject
+    Session defaultSession;
+
+    @Inject
+    @PersistenceUnit("users")
+    Session usersSession;
```

Comment:
ok that's interesting. Didn't expect to be able to combine `PersistenceUnit` on `Session`, but it's nice.

Reply:
This is Quarkus' own `PersistenceUnit` annotation, `io.quarkus.hibernate.orm.PersistenceUnit`. With the standard annotations you have to use `@PersistenceContext`; see `MultiplePersistenceUnitsResourceInjectionSessionTest` or `SinglePersistenceUnitResourceInjectionSessionTest`.